### PR TITLE
docs: Presign

### PR DESCRIPTION
**Pull request by soyeric128**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Added doc for the function PRESIGN

Fixes #issue

### CODE REVIEW

**Comment by Xuanwo** [comment]
Thanks for this PR!

File: docs/doc/30-reference/20-functions/123-file-functions/_category_.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "label": "File Functions",
```
Comment (Xuanwo) [inline docs/doc/30-reference/20-functions/123-file-functions/_category_.json:2]: This is not a function, it's a statement.

Place this statement under https://databend.rs/doc/reference/sql/ddl

File: docs/doc/30-reference/20-functions/123-file-functions/presign.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+Generates the pre-signed URL for a staged file by the stage name and file path you provide. The pre-signed URL enables you to access the file through a web browswer or an API request.
+
+:::tip
+This function currently works with files stored at Amazon S3 only.
+:::
+
+## Syntax
+
+```sql
+PRESIGN @<stage_name>/.../<file_name>
```
Comment (Xuanwo) [inline docs/doc/30-reference/20-functions/123-file-functions/presign.md:14]: The full syntax should be:

```sql
PRESIGN [ { DOWNLOAD | UPLOAD }] @<stage_name>/.../<file_name> [ EXPIRE = <expire_in_seconds> ]
```

File: docs/doc/30-reference/20-functions/123-file-functions/presign.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+```
+
+## Examples
+
+This example generates the pre-signed URL for the file `books.csv` on the stage `my-stage`:
```
Comment (Xuanwo) [inline docs/doc/30-reference/20-functions/123-file-functions/presign.md:19]: Please adds the following examples:

- `PRESIGN @my_stage/books.csv` get a download link
- `PRESIGN @my_stage/books.csv EXPIRE = 7200` get a download link which expired in 7200 seconds (2 hours)
- `PRESIGN DOWNLOAD @my_stage/books.csv` is the same with `PRESIGN @my_stage/books.csv`
- `PRESIGN UPLOAD @my_stage/books.csv` generate a link to upload as books.csv under `@my_stage`.

**Comment by Xuanwo** [comment]
More information about this SQL could be found at https://databend.rs/doc/contributing/rfcs/presign

File: docs/doc/30-reference/20-functions/123-file-functions/_category_.json (hunk 1 above)
Comment (soyeric128) [reply docs/doc/30-reference/20-functions/123-file-functions/_category_.json:2]: I saw SF lists it as a part of file functions. Is it different?
https://docs.snowflake.com/en/sql-reference/functions-file.html

File: docs/doc/30-reference/20-functions/123-file-functions/presign.md (hunk 2 above)
Comment (soyeric128) [reply docs/doc/30-reference/20-functions/123-file-functions/presign.md:14]: What's the default value for EXPIRE? 3,600 seconds?

File: docs/doc/30-reference/20-functions/123-file-functions/_category_.json (hunk 1 above)
Comment (Xuanwo) [reply docs/doc/30-reference/20-functions/123-file-functions/_category_.json:2]: Yep, we implement `presign` in a different way.

**Review by Xuanwo — APPROVED** [review]
Nice!

**Pull request merged**